### DISCUSSION TITLE
Ambiguous connectors on self intersecting/overlapping segments

### DISCUSSION THREAD

**Discussion by derolf** [OvertureMaps/data#282] [Discussion]
Example ONE:

Let's assume we have a single segment S that is shaped like this:

```
    -
   / \
   \ /
    X
   / \  
```

At the X, S intersects itself and a turn is allowed. So, is that modelled as a connector X with segment S listing X once in its connector_ids?

Example TWO:

Let's now assume that S is a spiral (like a spiral staircase or road) and let's assume that it connects to another segment T. A connector to connect S and T would have multiple positions on S on different levels/altitudes. How do we specify which one we want S and T to be connected at?

**Comment by brad-richardson** [comment]
For example one: S would be split at X into (at least) two new segments due to the self intersection at a connector which is never allowed. Depending on road class, I could see it being split into S1 (upper left, lower right) and S2 (upper right, lower left) which would no longer self intersect. Alternatively, four segments - upper/lower left/right.

For example two: Only knowing connector coordinates is a known deficiency the task force has been discussing. We're currently proposing to include the linear referenced value [0, 1] for each connector_id on a segment to remedy.

Let me know if I misunderstood anything!

**Reply by brad-richardson** [reply]
The linear reference value is now included on a new `connectors` property to disambiguate
